viewing_party/party.py

```diff
@@ -1,23 +1,187 @@
+        return avg
+
+
+def get_most_watched_genre(user_data):
+    get_most_watched = []
+    if bool(user_data["watched"]) == False:
+        get_most_watched.append(None)
+        popular_genre = statistics.mode(get_most_watched)
+        return popular_genre
+        
+    else:
+        for movie in user_data["watched"]:
+            get_most_watched.append(movie["genre"])
+            popular_genre = statistics.mode(get_most_watched)
+        return popular_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+# get list of all friends movies(titles)
+# check if each users movie is in list of movie(title)
+
+def get_unique_watched(user_data):
+    unique_list = []
+    friends_titles= []
+
+    for friends_data in user_data["friends"]:
+        for friends_movie in friends_data["watched"]:
+            friends_titles.append(friends_movie["title"])
+
+
+    for movie in user_data["watched"]:
+        if movie["title"] not in friends_titles:
+            unique_list.append(movie)
+
+    return unique_list
+
+def get_friends_unique_watched(user_data):
+    friends_movie_list =[]
+    for friend_data in user_data["friends"]:
+        for friends_movie in friend_data["watched"]:
+            if friends_movie not in user_data["watched"] and friends_movie not in friends_movie_list:
+                friends_movie_list.append(friends_movie)
+    return friends_movie_list
+
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommendations = []
+
+    unique_watch_list = get_friends_unique_watched(user_data)
+    
+    for user_sub in range(len(user_data["subscriptions"])):
+        for sub in range(len(unique_watch_list)):
+            if user_data["subscriptions"][user_sub] == unique_watch_list[sub]["host"]:
+                recommendations.append(unique_watch_list[sub])
+
+
+    return recommendations
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    
+    mode_genre = get_most_watched_genre(user_data)
+    new_recommendations = []
+    rec_recommendations = get_friends_unique_watched(user_data)
+
+    for user_movie in rec_recommendations:
+        if rec_recommendations[rec_recommendations.index(user_movie)]["genre"] == mode_genre:
+            new_recommendations.append(user_movie)
+
+    return new_recommendations
+
+
+
+def get_rec_from_favorites(user_data):
+    rec_recommendations = []
+    unique_watch_list = get_unique_watched(user_data)
+
+    for user_movie in unique_watch_list:
+        if user_movie in user_data["favorites"]:
+            rec_recommendations.append(user_movie)
+
+    return rec_recommendations
+
+
+
+
+    #refactored code 

```

Comment:
I would move this commented code to your notes or a txt file for your reference. You wouldn't want this to be in your final code.